src/_pytest/mark/structures.py

```diff
@@ -153,8 +161,19 @@ def combined_with(self, other):
         combines by appending args and merging the mappings
         """
         assert self.name == other.name
+
+        param_ids_from = None  # type: Optional[Mark]
+        if self.name == "parametrize":
+            if other._has_param_ids():
+                param_ids_from = other
+            elif self._has_param_ids():
+                param_ids_from = self
+
         return Mark(
-            self.name, self.args + other.args, dict(self.kwargs, **other.kwargs)
+            self.name,
+            self.args + other.args,
+            dict(self.kwargs, **other.kwargs),
+            param_ids_from=param_ids_from,
         )
```

Comment:
I've thought about having a specialized `ParametrizedMark` to handle the extra logic here.
But it was not clear to me if `MarkDecorator` would be a better fit to begin with, and then the name check would be done when generating/instantiating marks instead.

Reply:
Object based markers would resolve that at mark creation time